railties/lib/rails/generators/devcontainer.rb

```diff
@@ -84,128 +78,180 @@ def devcontainer_needs_redis?
           !(options.skip_action_cable? && options.skip_active_job?)
         end
 
-        def db_port_for_devcontainer(database = options[:database])
-          case database
-          when "mysql", "trilogy" then 3306
-          when "postgresql"       then 5432
-          end
+        def devcontainer_database
+          @devcontainer_database ||= Database.build(options[:database])
         end
 
-        def db_name_for_devcontainer(database = options[:database])
-          case database
-          when "mysql"          then "mysql"
-          when "trilogy"        then "mariadb"
-          when "postgresql"     then "postgres"
-          end
+        def devcontainer_db_service_yaml(**options)
+          return unless service = devcontainer_database.service
+
+          { devcontainer_database.name => service }.to_yaml(**options)[4..-1]
         end
 
-        def db_volume_name_for_devcontainer(database = options[:database])
-          case database
-          when "mysql"          then "mysql-data"
-          when "trilogy"        then "mariadb-data"
-          when "postgresql"     then "postgres-data"
-          end
+        def local_rails_mount
+          {
+            type: "bind",
+            source: Rails::Generators::RAILS_DEV_PATH,
+            target: Rails::Generators::RAILS_DEV_PATH
+          }
         end
 
-        def db_package_for_dockerfile(database = options[:database])
-          case database
-          when "mysql"          then "default-libmysqlclient-dev"
-          when "postgresql"     then "libpq-dev"
+        class Database
```

Comment:
Can we move this entire class to railties/lib/rails/generators/database.rb? There is a lot of duplication still on that file that we can remove now that we have a abstraction to the database on generators.

Reply:
Yes this is still WIP

Reply:
@rafaelfranca Alright ready for another review